src/boot/efi/boot.c

```diff
@@ -1532,6 +1541,34 @@ static void config_entry_add_type1(
         TAKE_PTR(entry);
 }
 
+static EFI_STATUS efivar_get_timeout(const char16_t *var, uint32_t *ret_value) {
+        _cleanup_free_ char16_t *value = NULL;
+        EFI_STATUS err;
+
+        assert(var);
+        assert(ret_value);
+
+        err = efivar_get(LOADER_GUID, var, &value);
+        if (err != EFI_SUCCESS)
+                return err;
+
+        if (streq16(value, u"menu-force")) {
+                *ret_value = TIMEOUT_MENU_FORCE;
+                return EFI_SUCCESS;
+        }
+        if (streq16(value, u"menu-hidden")) {
+                *ret_value = TIMEOUT_MENU_HIDDEN;
+                return EFI_SUCCESS;
+        }
```

Comment:
So, I am not convinced about this commit. This should be split out at the very least.

The thing is that "bootctl" and "sd-boot" might not be updated in lock-step (after all sd-boot is supposed to be shared between multiple OSes on the same host, and thus you might end up with sd-boot from fedora and a bootctl from suse in a very different version). So just setting these values from userspace and expecting them to work is not going to fly...

We communicate boot loader features from sd-boot to userspace via LoaderFeatures efi var. This is how we usually avoid such incompatibilities. But for this thing here it appears to be a bit too much.

Moroever, this stuff is actually defined as part of https://systemd.io/BOOT_LOADER_INTERFACE, i.e. is supposed to be stable...

Reply:
You know, I honestly think you're deluding yourself a bit on the BLS. I am actually unaware of any other boot loader implementing it after all these years (aside from some attempt by fedora for grub which afaik is bit-rotting and not upstreamed?).

Also, sd-boot and the BLS are already at odds with each other on this matter: `The time is specified in µs.`. sd-boot and bootctl always used seconds here.

Reply:
oh, the interface spec is IRL not so much about other boot loaders implementing it. It's more about other userspace tools making use of it. I happen to know some projects which set the efi vars in their own code and assume things just work. which is irrelevant for the issue at hand here, but it still makes the spec useful in general. And the fact it's official independent of a specific implementation (though in reality it's about sd-boot) makes it easier politicically for 3rd parties to support it without communicating so much they are binding things to systemd (even though they do)

Reply:
btw if you are talking about the boot loader spec (not the boot loader interface): fedora and others actually *do* implement that and use it by default in grub. I think they are working to get that upstream too.